packages/marko-web/components/page/metadata/google-structured-data/content.js

```diff
@@ -20,58 +20,93 @@ const getImages = (node) => {
-      },
+  if (node.type === 'podcast') {
+    // Set associatedMedia if audio file is present
+    const associatedMedia = (get(node, 'fileName') && get(node, 'fileSrc'))
+      ? {
+        '@type': 'AudioObject',
+        contentUrl: get(node, 'fileSrc'),
+        name: get(node, 'fileName'),
+      }
+      : undefined;
+    return JSON.stringify({
+      ...defaultStruturedData,
+      '@type': 'PodcastEpisode',
       headline: get(node, 'metadata.title'),
-      image: getImages(node),
-      datePublished: publishedISOString,
-      dateModified: updatedISOString,
-      author: getAuthor(node),
-      description: get(node, 'metadata.description'),
+      ...(associatedMedia && { associatedMedia }),
+    });
+  }
+
+  if (node.type === 'company') {
+    // Concat address1 && address2 if present
+    const streetAddresses = [];
+    if (get(node, 'address1')) streetAddresses.push(get(node, 'address1'));
+    if (get(node, 'address2')) streetAddresses.push(get(node, 'address2'));
+    const streetAddress = streetAddresses.join(' ');
+    // Display and set address info if present
+    const showAddress = (streetAddress || get(node, 'city') || get(node, 'state') || get(node, 'zip'));
+    const address = (showAddress)
+      ? {
+        '@type': 'PostalAddress',
+        ...(get(node, 'city') && { addressLocality: get(node, 'city') }),
+        ...(get(node, 'state') && { addressLocality: get(node, 'state') }),
+        ...(get(node, 'zip') && { postalCode: get(node, 'zip') }),
+        ...(streetAddress && { streetAddress }),
+      }
+      : undefined;
+    // Set tollfree || phone if present
+    const telephone = get(node, 'tollfree') || get(node, 'phone');
+
+    return JSON.stringify({
+      ...defaultStruturedData,
+      '@type': 'Organization',
+      address,
+      ...(telephone && { telephone }),
+      ...(get(node, 'metadata.image.src') && { logo: get(node, 'metadata.image.src') }),
+      ...(get(node, 'email') && { email: get(node, 'email') }),
+      ...(get(node, 'fax') && { faxNumber: get(node, 'fax') }),
     });
-    return structuredData;
   }
 
-  const standardArticleTypes = ['product', 'blog'];
-  if (standardArticleTypes.includes(node.type)) {
-    const structuredData = JSON.stringify({
-      '@context': 'https://schema.org',
-      '@type': 'Article',
-      mainEntityOfPage: {
-        '@type': 'WebPage',
-        '@id': get(node, 'siteContext.canonicalUrl'),
-      },
+  if (['article', 'news'].includes(node.type)) {
+    return JSON.stringify({
+      ...defaultStruturedData,
+      '@type': 'NewsArticle',
       headline: get(node, 'metadata.title'),
-      image: getImages(node),
-      datePublished: publishedISOString,
-      dateModified: updatedISOString,
-      author: getAuthor(node),
-      description: get(node, 'metadata.description'),
     });
-    return structuredData;
   }
 
-  return undefined;
+  return JSON.stringify(defaultStruturedData);
```

Comment:
I still think we need to return undefined so we're not assuming structured data for other content types